server/s3_common_utilities.cc

```diff
@@ -179,3 +182,29 @@ std::string evhtp_error_flags_description(uint8_t errtype) {
 }
 
 }  // namespace S3CommonUtilities
+
+void s3_kickoff_graceful_shutdown(int ignore) {
+  extern int global_shutdown_in_progress;
```

Comment:
If time permits, we should make this variable as bool.. We are in C++ :)

Reply:
need to check with func author, will take some time